openverse_catalog/dags/maintenance/fix_phylopic_foreign_identifier.py

```diff
@@ -0,0 +1,116 @@
+"""
+One-time run DAG to fix the foreign identifier for PhyloPic images.
+
+In order to prevent broken links, we need to update the foreign identifier from using
+the image URL to using the foreign image UUID.
+"""
+import logging
+import re
+from textwrap import dedent
+
+from airflow.decorators import dag
+from airflow.models.abstractoperator import AbstractOperator
+from airflow.operators.python import PythonOperator
+from psycopg2.errors import UniqueViolation
+
+from common.constants import DAG_DEFAULT_ARGS, POSTGRES_CONN_ID, XCOM_PULL_TEMPLATE
+from common.slack import send_message
+from common.sql import PostgresHook
+
+
+logger = logging.getLogger(__name__)
+
+DAG_ID = "update_phylopic_foreign_identifier"
+
+
+def update_foreign_identifiers(task: AbstractOperator) -> dict[str, int]:
+    pg = PostgresHook(
+        postgres_conn_id=POSTGRES_CONN_ID,
+        default_statement_timeout=PostgresHook.get_execution_timeout(task),
+    )
+    phylopic_rows = pg.get_records(
+        "SELECT foreign_identifier, identifier FROM image WHERE provider = 'phylopic';"
+    )
+
+    uuid_pattern = (
+        "[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}"
+    )
```

Comment:
I think we could compile this pattern which might make the repeated regex a bit faster/more efficient: https://docs.python.org/3/library/re.html#re.compile